docs/tutorial/transforms.md

```diff
@@ -1,3 +1,193 @@
 # Transforms
 
-TODO
+Among arr.ai's powerful features, transform operators are perhaps the most
+heavily used. They come in four flavors, each operating on a different kind of
+input:
+
+- `->` transforms a single value.
+- `=>` transforms the members of a set.
+- `>>` transforms the entries of a sequence or dictionary.
+- `:>` transforms the attributes of a tuple.
+
+The `->` transform operator has the following syntactic structure:
+
+- `lhs -> expr` binds the name `.` to `lhs` and evaluates `expr` in this
+  context.
+- `lhs -> \pattern expr` binds `pattern` to `lhs` and evaluates `expr` in this
+  context.
+
+The other three operators follow the same syntax. However, they bind each of the
+elements of the `lhs`, not the entire `lhs`.
+
+If this isn't clear, don't worry, it'll become clear as we work through the
+examples below.
+
+## The basics
+
+Let's explore the basic usage of the above operators, one at a time.
+
+### `->`
+
+The arrow operator, `->`, can seem almost trivial:
+
+```arrai
+@> 42 -> .
+@> 42 -> . + 1
+@> "hello, " -> . ++ "world"
+```
+
+The alternative form is likewise fairly trivial:
+
+```arrai
+@> 42 -> \x x + 1
+```
+
+This operator shows its usefulness when `lhs` becomes more complex:
+
+```arrai
+@> /set data = (customer: (custid: 123, name: (first: "Franz", last: "Kafka")))
+@> data.customer.name -> $`${//str.upper(.last)}, ${.first}`
+```
+
+As you can see, `->` offers a very lightweight mechanism to refer to a common
+complex expression while allowing a reader to follow a natural left-to-right
+order as they seek to understand the intent.
+
+### `=>`
+
+The "set-arrow" operator, `=>`, requires `lhs` to be a set. It applies the
+transform to each member of `lhs` and evaluates to a set of the results:
+
+```arrai
+@> {2, 4, 6, 8, 10} => . + 1
+@> {2, 4, 6, 8, 10} => . % 6
+```
+
+Note that, if the transform produces the same output for some of the `lhs`
+elements, the resulting set will have fewer members than `lhs`.
```

Comment:
Maybe this would be a good chance to remind a new user that sets are different from arrays, and cannot contain duplicates?